src/Microsoft.VisualStudio.ProjectSystem.VisualBasic.VS/ProjectSystem/VS/Automation/VSImports.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.ComponentModel.Composition;
+using System.Linq;
+using EnvDTE;
+using Microsoft.VisualStudio.ProjectSystem.VS.ConnectionPoint;
+using VSLangProj;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Automation
+{
+    [Export(typeof(Imports))]
+    [Export(typeof(ImportsEvents))]
+    [AppliesTo(ProjectCapability.VisualBasic)]
+    internal class VSImports : ConnectionPointContainer,
```

Comment:
In legacy project, the VSImports maintains a list and updates the list during addition and deletion. When asked for a enumerator, it just returns an enumerator over the existing list. This has the same problem as here. 

CPS, on the other hand, maintains a ReferencesHostBridge object at the Hierarchy level, that coordinates the changes with Update Subscription and looks into all the ReferencesServices. 

We could have an object that subscribes to changes to the Imports, which provides the backend for the list. But given the importance of this workitem. I would open an issue and track this, since this approach is a overkill for this feature.